hrjobcontract/js/src/job-contract/controllers/form/form-leave.js

```diff
@@ -0,0 +1,38 @@
+define([
+    'job-contract/vendor/fraction',
+    'job-contract/controllers/controllers'
+], function (Fraction, controllers) {
+    'use strict';
+
+    controllers.controller('FormLeaveCtrl',['$scope', '$rootScope', '$filter', '$log', '$q', 'UtilsService',
+      function ($scope, $rootScope, $filter, $log, $q, UtilsService) {
+        $log.debug('Controller: FormLeaveCtrl');
+
+        var promiseNumberOfPublicHolidays = UtilsService.getNumberOfPublicHolidaysInCurrentPeriod();
+
+        promiseNumberOfPublicHolidays.then(function(result){
+          var number = parseFloat(result);
+          if(isNaN(number)) {
+            number = 0;
+          }
+          $scope.numberOfPublicHolidays = number;
+        });
+
+        $scope.$watch('entity.leave', function(newValue, oldValue) {
+          var newTypeWithPublicHolidays = null;
+          newValue.forEach(function(value, i) {
+            if(value.add_public_holidays === '1' && oldValue[i].add_public_holidays === '0') {
```

Comment:
That was indeed really ugly. The reason for this is part because of how MySQL stores the data and part because of how the API returns the data (as a string). I fixed this by updating the service methods that fetch the data to covert them to booleans.